services/bitbucket-server/bitbucket-server.service.js

```diff
@@ -0,0 +1,77 @@
+'use strict'
+
+const BaseJsonService = require('../base-json')
+const serverSecrets = require('../../lib/server-secrets')
+
+const Joi = require('joi')
+const schema = Joi.object({
+  size: Joi.number()
+    .integer()
+    .min(0)
+    .required(),
+}).required()
+
+module.exports = class BitbucketServer extends BaseJsonService {
+  static get category() {
+    return 'build'
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'Open Pull Requests',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Bitbucket Server: Open Pull Requests',
+        namedParams: {
+          proto: 'https',
+          host: 'bitbucket.mydomain.net',
+          project: 'foo',
+          repo: 'bar',
+        },
+        staticExample: this.render({ count: 7 }),
+        keywords: ['bitbucket', 'pull-request'],
+      },
+    ]
+  }
+
+  static get route() {
+    return {
+      base: 'bitbucket-server',
+      pattern: ':proto/:host/:project/:repo/pr',
+    }
+  }
+
+  static render({ count }) {
+    return {
+      message: count.toString(),
+      color: 'brightgreen',
+    }
+  }
+
+  async fetch({ proto, host, project, repo }) {
+    let authenticatedHost = host
+    if (
+      serverSecrets &&
+      serverSecrets.bitbucket_server_username &&
+      serverSecrets.bitbucket_server_password
+    ) {
+      authenticatedHost = `${serverSecrets.bitbucket_server_username}:${
+        serverSecrets.bitbucket_server_password
+      }@${host}`
+    }
+
+    return this._requestJson({
+      schema,
+      url: `${proto}://${authenticatedHost}/rest/api/1.0/projects/${project}/repos/${repo}/pull-requests?limit=100&state=OPEN&wthProperties=false&withAttributes=false`,
```

Comment:
Also are there are any typical HTTP response codes we could see from BBS (404, etc.)? For example when the specified `repo` doesn't exist in the specified `project`, specified `project` doesn't exist, not authorized to access specified repo and/or project, BBS instance isn't reachable, etc. 

If there then we can utilize the `errorMessages` param to the `_requestJson()` function to provide a more meaningful error message for the user. For example:

```js
errorMessage: {
  404: 'project or repo not found`,
  403: 'not authorized to access project or repo',
}
```

Reply:
I'll have to try some examples and/or dig through their documentation but there are most likely some codes we could watch out for.

Reply:
No worries if there aren't. I've used BBS before but never the APIs directly. I personally use the Advanced Rest Client frequently when working on a service as I find it helps me check the responses (codes and headers in particular) very easily